core/src/test/java/io/parsingdata/metal/AutoEqualityTest.java

```diff
@@ -313,31 +316,38 @@ private static Class<?> getClass(final String className) {
     private static Collection<Arguments> generateObjectArrays(final Set<Class<?>> classes) throws IllegalAccessException, InstantiationException, InvocationTargetException {
         Collection<Arguments> results = new ArrayList<>();
         for (Class<?> c : classes) {
-            results.add(generateObjectArrays(c));
+            results.addAll(generateObjectArrays(c));
         }
         return results;
     }
 
-    private static Arguments generateObjectArrays(final Class<?> c) throws IllegalAccessException, InvocationTargetException, InstantiationException {
-        Constructor<?> cons = c.getDeclaredConstructors()[0];
-        cons.setAccessible(true);
-        List<List<Supplier<Object>>> args = new ArrayList<>();
-        for (Class<?> cl : cons.getParameterTypes()) {
-            if (!mapping.containsKey(cl)) {
-                throw new AssertionError("Please add a mapping for type " + cl.getSimpleName());
+    private static List<Arguments> generateObjectArrays(final Class<?> c) throws IllegalAccessException, InvocationTargetException, InstantiationException {
+        final List<Arguments> arguments = new ArrayList<>();
+        for (Constructor<?> cons : c.getDeclaredConstructors()) {
+            final boolean containsGenericArgument = Arrays.stream(cons.getParameterTypes()).anyMatch(p -> p == Object.class);
+            if (containsGenericArgument) {
+                break;
             }
-            args.add(mapping.get(cl));
-        }
-        List<List<Supplier<Object>>> argLists = generateCombinations(0, args);
-        List<Object> otherInstances = new ArrayList<>();
-        for (List<Supplier<Object>> argList : argLists.subList(1, argLists.size())) {
-            otherInstances.add(cons.newInstance(instantiate(argList).toArray()));
+            cons.setAccessible(true);
+            List<List<Supplier<Object>>> args = new ArrayList<>();
+            for (Class<?> cl : cons.getParameterTypes()) {
+                if (!mapping.containsKey(cl)) {
+                    throw new AssertionError("Please add a mapping for type " + cl.getSimpleName());
+                }
+                args.add(mapping.get(cl));
+            }
+            List<List<Supplier<Object>>> argLists = generateCombinations(0, args);
+            List<Object> otherInstances = new ArrayList<>();
+            for (List<Supplier<Object>> argList : argLists.subList(1, argLists.size())) {
```

Comment:
Another missing `final` here.

```suggestion
            for (final List<Supplier<Object>> argList : argLists.subList(1, argLists.size())) {
```